pkg/database/user_report.go

```diff
@@ -110,13 +120,75 @@ func (db *Database) FindUserReport(tx *gorm.DB, phoneNumber string) (*UserReport
 
 // DeleteUserReport removes a specific phone number from the user report
 // de-duplication table.
-func (db *Database) DeleteUserReport(phoneNumber string) error {
+// The actor may be nil on this operation, if it's the result of a
+// natural delete (SMS failure).
+// Otherwise this is an administrative action.
+func (db *Database) DeleteUserReport(phoneNumber string, actor Auditable) error {
 	hmacedCodes, err := db.generatePhoneNumberHMACs(phoneNumber)
 	if err != nil {
 		return fmt.Errorf("failed to create hmac: %w", err)
 	}
 
-	return db.db.Where("phone_hash IN (?)", hmacedCodes).Delete(&UserReport{}).Error
+	return db.db.Transaction(func(tx *gorm.DB) error {
+		var ur UserReport
+		if err := tx.
+			Set("gorm:query_option", "FOR UPDATE").
+			Model(&UserReport{}).
+			Where("phone_hash IN (?)", hmacedCodes).
+			First(&ur).
+			Error; err != nil {
+			if IsNotFound(err) {
```

Comment:
We tend to just return a wrapped error here and let the caller handle whether it's `NotFound` or `ValidationError` in the controller (since it could indicate a different response).